### DISCUSSION TITLE
Adding VOC device class for Air Quality sensors

### DISCUSSION THREAD

**Discussion by reefab** [home-assistant/architecture#608] [Entity Models] [answered]
In the previous discussion about adding a device class for air quality monitors: #109

There is a mention to have 'volatile_organic_compounds' as an attribute but alas, it is not present in the component.

I'm currently working to migrate the foobot sensor to this air quality platform and I figured this would make sense to have it part of the generic component as many indoor air quality monitors will provide this measurement. Especially considering that many of those monitors do not have an actual CO2 sensor but derives CO2 values from their VOC sensor.

I already made a PR request adding this attribute and the corresponding property to the component: https://github.com/home-assistant/home-assistant/pull/20306 but I'm opening this issue as requested. I'll make another PR if there is an agreement.

**Comment by Jc2k** [comment]
This has come up as part of https://github.com/home-assistant/home-assistant/pull/30383. @itewk pointed out there is a discrepancy between the docs [here](https://developers.home-assistant.io/docs/en/entity_air_quality.html) (which lists VOC) and the code [here](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/air_quality/__init__.py) (which doesn't).

I'd certainly support adding it to the base entity. It looks like the dyson fans already support it and both homekit integrations want to use the attribute. It looks like xiaomi_miio may have support as well, but with a different attribute name.

**Comment by ahayworth** [comment]
This would be useful for the Awair component as well, which reports a "total VOCs" measurement (but does not break them out individually). If there is consensus that `volatile_organic_compounds` should be added to the base component, I will be happy to submit a PR.

**Comment by Jc2k** [comment, marked as answer]
This topic was originally raised in the context of the `air_quality` platform, which is now deprecated.

However I think it is a valid sensor value to consider for a device class (as per https://github.com/home-assistant/core/pull/54075) and I would like to add this to homekit_controller (as per https://github.com/home-assistant/core/pull/54673).

Like with those, if there is a DEVICE_CLASS then integrations like homekit bridge can automatically expose the right sensor type when exporting the entity back out to iOS/Siri.

I'd propose `µg/m³` as the unit, as that is what the existing voc sensors seem to support (where there is documentation).

**Reply by MartinHjelmare** [reply]
Can we estimate how many integrations could take advantage of this new device class?

**Reply by itewk** [reply]
at the very least all the dyson fans.

**Reply by Jc2k** [reply]
There are 5 integrations which already reference volatile_organic_compounds. (homekit_controller, dyson, awair, kaiterra, xiaomi_miio).

 I think there are a bunch more (3+) referencing `voc`, but thats a bit annoying to grep for so might have missed some. (airvisual, foobar, zha)

`homekit` doesn't use it currently, but it could make use of this devce class. E.g. could automatically create the right characteristics in iOS for sensors tagged with this.

Imagine mqtt/esphome fall into that category as well?

**Reply by MartinHjelmare** [reply]
Thanks! With that number of integrations I'm in favor of adding this device class 👍.

**Reply by frenck** [reply]
Agree on adding as well. ESPHome can also use/support this.

**Reply by Jc2k** [reply]
PR for this is here: https://github.com/home-assistant/core/pull/55050
